src/ddmd/e2ir.d

```diff
@@ -194,27 +194,21 @@ elem *callfunc(Loc loc,
                 arg.error("simd operator must be an integer constant, not '%s'", arg.toChars());
         }
 
-        for (size_t i = 0; i < arguments.dim; i++)
-        {
-        Lagain:
-            Expression arg = (*arguments)[i];
-            assert(arg.op != TOKtuple);
-            if (arg.op == TOKcomma)
-            {
-                CommaExp ce = cast(CommaExp)arg;
-                eside = el_combine(eside, toElem(ce.e1, irs));
-                (*arguments)[i] = ce.e2;
-                goto Lagain;
-            }
-        }
+        /* Convert arguments[] to elems[] in left-to-right order
+         */
+        const n = arguments.dim;
+        elem*[5] elems_array = void;
+        import core.stdc.stdlib : malloc, free;
+        auto elems = (n <= 5) ? elems_array.ptr : cast(elem**)malloc(arguments.dim * (elem*).sizeof);
+        assert(elems);
```

Comment:
I've avoided that form because the intervening code is not `nothrow`, meaning it sets up an EH frame even though the intervening code will never throw.